buffer/src/main/java/io/netty/buffer/UnpooledByteBufAllocator.java

```diff
@@ -86,13 +87,7 @@ protected ByteBuf newHeapBuffer(int initialCapacity, int maxCapacity) {
 
     @Override
     protected ByteBuf newDirectBuffer(int initialCapacity, int maxCapacity) {
-        final ByteBuf buf;
-        if (PlatformDependent.hasUnsafe()) {
-            buf = noCleaner ? new InstrumentedUnpooledUnsafeNoCleanerDirectByteBuf(this, initialCapacity, maxCapacity) :
-                    new InstrumentedUnpooledUnsafeDirectByteBuf(this, initialCapacity, maxCapacity);
-        } else {
-            buf = new InstrumentedUnpooledDirectByteBuf(this, initialCapacity, maxCapacity);
-        }
+        final ByteBuf buf = new InstrumentedUnpooledDirectByteBuf(this, initialCapacity, maxCapacity);
```

Comment:
I don't think this is correct... we still want to return an instance that uses `Unsafe` if we can access Unsafe. Otherwise we will loose other optimizations.